src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -285,10 +285,12 @@ export class Kernel implements IKernel {
         if (isPythonKernelConnection(this.kernelConnectionMetadata)) {
             await this.notebook.setLaunchingFile(this.uri.fsPath);
         }
-        await this.notebook
-            .requestKernelInfo()
-            .then((item) => (this._info = item.content))
-            .catch(traceWarning.bind('Failed to request KernelInfo'));
+        if (!this._info) {
```

Comment:
See comments in the original issue, updated with details.

Reply:
i think we should remove this new if condition. If users switch/restart kernels, we'd be using the old kernel information, which would be incorrect.
I.e. this method should never cache it.